### DISCUSSION TITLE
Does the script handle duplicate file names?

### DISCUSSION THREAD

**Discussion by GuoxiongXie** [TheLastGimbus/GooglePhotosTakeoutHelper#244] [I NEED HELP]
First of all, thanks for the script!
My Google Photos Takeout contains multiple photos that have the same name in different zip folders (as a result, they also have the same name for metadata json files).

Question 1: I'm wondering if the script will handle those files with the same names, i.e. keep them all in one folder without overwriting/replacing each other (maybe append a count/random string to the end of the file when duplicate names are found). 

Question 2: If the script does handle file name collision, what's the scripts strategy of matching the photos with metadata json files. For example, let's say there are two files with the same name: first photo and second photo; and there are two metadata json with the same name: first photo json and second photo json. How does the script know it should match first photo json with first photo, but not match second photo json with first photo?

Thanks a lot for your help!

**Comment by TheLastGimbus** [comment]
> the script will handle those files with the same names

yes it does :tada: - i can't guarantee anything for sure, because there are 85023952 cases for everything, but i can proudly say `gpth` handles this stuff *very good* - it will not copy same duplicate photo twice just because it has different name, nor will it lose your photos just because they do have same name - just run it and you will see!

> (maybe append a count/random string to the end)

It just adds `...(1)` at the end :+1:

> My Google Photos Takeout contains multiple photos that have the same name in different zip folders

If they will literally be same photos in same folders... then it's up to your unzipping app what to do about this :shrug: read the visual instruction in the readme

> what's the scripts strategy of matching the photos with metadata

What strategy? A smart one :brain:

Serious ansewar: in what folder they are. There cannot be two same named files in one folder. If there are in two different then it will just see that 

Even more serious answear - there are 115 lines of code dedicated to match those jsons, in all fucked up edge cases that we found: https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/blob/master/lib/date_extractors/json_extractor.dart
